### PR TITLE
feat: algokit debugger support

### PR DESCRIPTION
**Pull request by aorumbayev**
## Proposed changes

- Added algokit debugger support (expanding pyproject) 
- Added code tour getting started guide for the template
- Updating readme with instructions on enabling debugger and reference to main doc

## TODO

- [x] Patch up tests (its partially working now given that its not supporting the new debug utils on typescript variations yet)
- [x] Can't do so for utils-ts alternative until ts is at least prereleased on npm as our package.json on utils is not optimized for direct installs from github
- [x] No changes needed on react (as new debug utils require node and fs access)
- [x] Once this is merged we will just need a simple refresh pr on fullstack - no major changes on fullstack needed.

### CODE REVIEW

File: tests_generated/test_deployment_language-python/.copier-answers.yml [hunk 1]
```diff
@@ -8,19 +8,12 @@ author_email: None
 author_name: None
 contract_name: hello_world
 deployment_language: python
-ide_jetbrains: false
```
Comment (robdmoore) [inline tests_generated/test_deployment_language-python/.copier-answers.yml:11]: I assume this is caused by newer version of copier?

File: template_content/smart_contracts/__main__.py.jinja [hunk 2]
```diff
@@ -12,6 +12,11 @@ from smart_contracts.helpers.deploy import deploy
 from smart_contracts.helpers.build import build
 {%- endif %}
 
+# # Uncomment the following lines to enable auto generation of AVM Debugger compliant sourceMap and simulation trace file.
+# # Learn more about using AlgoKit AVM Debugger to debug your TEAL source codes and inspect varioud kidns of Algorand transactions in atomic groups -> https://github.com/algorandfoundation/algokit-avm-vscode-debugger
```
Comment (neilcampbell) [inline template_content/smart_contracts/__main__.py.jinja:16]:
```suggestion
# # Learn more about using AlgoKit AVM Debugger to debug your TEAL source codes and inspect various kinds of Algorand transactions in atomic groups -> https://github.com/algorandfoundation/algokit-avm-vscode-debugger
```

File: template_content/smart_contracts/{% if deployment_language == 'typescript' %}index.ts{% endif %}.jinja [hunk 3]
```diff
@@ -3,8 +3,13 @@ import * as path from 'path'
 import { consoleLogger } from '@algorandfoundation/algokit-utils/types/logging'
 import * as algokit from '@algorandfoundation/algokit-utils'
 
+// Uncomment the debug and traceAll options to enable auto generation of AVM Debugger compliant sourceMap and simulation trace file.
+// Learn more about using AlgoKit AVM Debugger to debug your TEAL source codes and inspect varioud kidns of Algorand transactions in atomic groups -> https://github.com/algorandfoundation/algokit-avm-vscode-Debugger
```
Comment (neilcampbell) [inline template_content/smart_contracts/{% if deployment_language == 'typescript' %}index.ts{% endif %}.jinja:7]:
```suggestion
// Learn more about using AlgoKit AVM Debugger to debug your TEAL source codes and inspect various kinds of Algorand transactions in atomic groups -> https://github.com/algorandfoundation/algokit-avm-vscode-Debugger
```

File: template_content/{% if ide_vscode %}.vscode{% endif %}/extensions.json.jinja [hunk 4]
```diff
@@ -16,6 +16,8 @@
     "esbenp.prettier-vscode",
     {% endif -%}
     "tamasfe.even-better-toml",
-    "editorconfig.editorconfig"
+    "editorconfig.editorconfig",
+    "vsls-contrib.codetour",
+    "algorand foundation.algokit-avm-vscode-debugger"
```
Comment (neilcampbell) [inline template_content/{% if ide_vscode %}.vscode{% endif %}/extensions.json.jinja:21]:
```suggestion
    "algorandfoundation.algokit-avm-vscode-debugger"
```
In theory it should be this.

File: tests_generated/test_default_parameters/.vscode/extensions.json [hunk 5]
```diff
@@ -3,6 +3,8 @@
     "ms-python.python",
     "ms-python.black-formatter",
     "tamasfe.even-better-toml",
-    "editorconfig.editorconfig"
+    "editorconfig.editorconfig",
+    "vsls-contrib.codetour",
+    "algorand foundation.algokit-avm-vscode-debugger"
```
Comment (neilcampbell) [inline tests_generated/test_default_parameters/.vscode/extensions.json:8]:
```suggestion
    "algorandfoundation.algokit-avm-vscode-debugger"
```

**Review by neilcampbell — APPROVED** [review]

**Pull request merged**